include/deal.II/base/symmetric_tensor.h

```diff
@@ -606,35 +614,27 @@ class SymmetricTensor
   /**
    * Add another tensor.
    */
-  SymmetricTensor &operator += (const SymmetricTensor &);
+  template <typename OtherNumber>
+  SymmetricTensor &operator += (const SymmetricTensor<rank,dim,OtherNumber> &);
 
   /**
    * Subtract another tensor.
    */
-  SymmetricTensor &operator -= (const SymmetricTensor &);
+  template <typename OtherNumber>
+  SymmetricTensor &operator -= (const SymmetricTensor<rank,dim,OtherNumber> &);
 
   /**
    * Scale the tensor by <tt>factor</tt>, i.e. multiply all components by
    * <tt>factor</tt>.
    */
-  SymmetricTensor &operator *= (const Number factor);
+  template <typename OtherNumber>
+  SymmetricTensor &operator *= (const OtherNumber &factor);
 
   /**
    * Scale the vector by <tt>1/factor</tt>.
    */
-  SymmetricTensor &operator /= (const Number factor);
-
-  /**
-   * Add two tensors. If possible, you should use <tt>operator +=</tt> instead
-   * since this does not need the creation of a temporary.
-   */
-  SymmetricTensor   operator + (const SymmetricTensor &s) const;
```

Comment:
Thats right. This now mimics the `Tensor` class (see [here](https://github.com/dealii/dealii/blob/master/include/deal.II/base/tensor.h#L1355)), and is correct because the return type of the operation is not necessarily equal to that of the LHS operand.